### PR TITLE
Certificates:  validate packages against signing requirements

### PR DESCRIPTION
**Pull request by dtivel**
Resolve https://github.com/NuGet/NuGetGallery/issues/5635.

### CODE REVIEW

File: src/Validation.Common.Job/Validation.Common.Job.csproj [hunk 1]
```diff
@@ -100,10 +100,10 @@
       <Version>2.23.0</Version>
     </PackageReference>
     <PackageReference Include="NuGet.Services.Validation">
-      <Version>2.23.0</Version>
+      <Version>2.25.0-master-29664</Version>
     </PackageReference>
     <PackageReference Include="NuGetGallery.Core">
-      <Version>4.4.4-dev-26726</Version>
+      <Version>4.4.4-dev-dtivel-certificates-29749</Version>
```
Comment (skofman1) [inline src/Validation.Common.Job/Validation.Common.Job.csproj]: >dev-dtivel-certi [](start = 21, length = 16)

you will change to an "official" version before merge, right?

Comment (dtivel) [reply src/Validation.Common.Job/Validation.Common.Job.csproj]: Yes.

File: src/Validation.PackageSigning.ProcessSignature/Job.cs [hunk 2]
```diff
@@ -30,7 +30,12 @@ protected override void ConfigureJobServices(IServiceCollection services, IConfi
 
             services.AddTransient<ISubscriptionProcessor<SignatureValidationMessage>, SubscriptionProcessor<SignatureValidationMessage>>();
 
-            services.AddTransient<IEntityRepository<Certificate>, EntityRepository<Certificate>>();
+            services.AddScoped<IEntitiesContext>(serviceProvider =>
```
Comment (skofman1) [inline src/Validation.PackageSigning.ProcessSignature/Job.cs]: Should be Transient. Scoped is per request, and there are no requests here since it's a job

Comment (loic-sharma) [reply src/Validation.PackageSigning.ProcessSignature/Job.cs]: Actually, Service Bus jobs create a scope for each message. See: https://github.com/NuGet/NuGet.Jobs/blob/245aa36f6b0d42a50a9c40a628149d29b5d653a8/src/Validation.PackageSigning.ProcessSignature/Job.cs#L86

File: src/Validation.PackageSigning.ProcessSignature/SignatureValidator.cs [hunk 3]
```diff
@@ -71,20 +71,28 @@ public class SignatureValidator : ISignatureValidator
...
+                return await HandleUnsignedPackageAsync(context);
             }
         }
-        
+
         private async Task<SignatureValidatorResult> HandleUnsignedPackageAsync(Context context)
         {
+            if (_corePackageService.IsSigningRequired(context.Message.PackageId))
+            {
+                _logger.LogInformation(
```
Comment (loic-sharma) [inline src/Validation.PackageSigning.ProcessSignature/SignatureValidator.cs]: >LogInformation [](start = 24, length = 14)

I would make this a `Warning`.

File: src/Validation.PackageSigning.ProcessSignature/SignatureValidator.cs [hunk 4]
```diff
@@ -71,20 +71,28 @@ public class SignatureValidator : ISignatureValidator
...
-                }
+
+                return await HandleUnsignedPackageAsync(context);
             }
         }
-        
+
         private async Task<SignatureValidatorResult> HandleUnsignedPackageAsync(Context context)
         {
+            if (_corePackageService.IsSigningRequired(context.Message.PackageId))
```
Comment (loic-sharma) [inline src/Validation.PackageSigning.ProcessSignature/SignatureValidator.cs]: >IsSigningRequired [](start = 36, length = 17)

Can the core package service methods be asynchronous?

File: src/Validation.PackageSigning.ProcessSignature/Job.cs (hunk 2 above)
Comment (skofman1) [reply src/Validation.PackageSigning.ProcessSignature/Job.cs]: Didn't know that. Thanks Loic!

**Pull request merged**